PowerShellTools/DebugEngine/Remote/RemoteEnumDebugProcess.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Debugger.Interop;
+using Automation = System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Diagnostics;
+using System.Windows.Forms;
+using PowerShellTools.Common;
+
+namespace PowerShellTools.DebugEngine.Remote
+{
+    /// <summary>
+    /// Works with the host service to find all attachable processes on a 
+    /// remote machine. Stores said processes in a list structure.
+    /// </summary>
+    internal class RemoteEnumDebugProcess : IEnumDebugProcesses2
+    {
+        private List<ScriptDebugProcess> _runningProcesses;
+        private string _remoteComputer;
+        private uint _currIndex;
+
+        public RemoteEnumDebugProcess(string remoteComputer)
+        {
+            _runningProcesses = new List<ScriptDebugProcess>();
+            _remoteComputer = remoteComputer;
+            _currIndex = 0;
+        }
+
+        /// <summary>
+        /// Asks HostService to find all attachable processes on the given machine. Will prompt user to retry connecting if 
+        /// the call to EnumerateRemoteProcesses returns null.
+        /// </summary>
+        /// <param name="remotePort"></param>
+        public void connect(IDebugPort2 remotePort)
+        {
+            // host needs to be initialized before we can connect/enumerate
+            UiContextUtilities.ActivateUiContext(UiContextUtilities.CreateUiContext(Constants.PowerShellReplCreationUiContextGuid));
+            if (!PowerShellToolsPackage.PowerShellHostInitialized)
+            {
+                // TODO: UI Work required to give user inidcation that it is waiting for debugger to get alive.

```

Comment:
Not sure if this should be some VS decision to make, but if you think there are some work can be done by us, please go enter an issue on github and add the issue number here in the comment.